spotipy/cache_handler.py

```diff
@@ -107,3 +108,45 @@ def get_cached_token(self):
 
     def save_token_to_cache(self, token_info):
         self.token_info = token_info
+
+
+class RedisCacheHandler(CacheHandler):
+    """
+    A cache handler that stores the token info in the Redis.
+    """
+
+    def __init__(self, redis, key=None):
+        """
+        Parameters:
+            * redis: Redis object provided by redis-py library
+            (https://github.com/redis/redis-py)
+            * key: May be supplied, will otherwise be generated
+                   (takes precedence over `token_info`)
+        """
+        self.redis = redis
+        self.key = key if key else 'token_info'
+
+        try:
+            from redis import RedisError
+        except ImportError:
+            warnings.warn('Error importing requied module "redis"', UserWarning)
```

Comment:
There's a typo here: "requied".

Reply:
Also, the word "required" may imply to some users that this module is required for the whole package. Instead, the error message should indicate that this module is required *for the `RedisCacheHandler`*.

Reply:
Fixed. I also restructured the history to preserve the evolution of Redis support on the master branch and diminish the extent of my own changes.